elixir/apps/domain/lib/domain/replication/manager.ex

```diff
@@ -18,23 +17,24 @@ defmodule Domain.Replication.Manager do
 
   @impl true
   def init(connection_module) do
+    Process.flag(:trap_exit, true)
     send(self(), {:connect, connection_module})
-
-    {:ok, %{retries: 0}}
+    {:ok, %{retries: 0, connection_pid: nil, connection_module: connection_module}}
   end
 
   @impl true
   def handle_info({:connect, connection_module}, %{retries: retries} = state) do
     Process.send_after(self(), {:connect, connection_module}, @retry_interval)
 
     case connection_module.start_link(replication_child_spec(connection_module)) do
-      {:ok, _pid} ->
-        # Our process won
-        {:noreply, %{state | retries: 0}}
+      {:ok, pid} ->
+        {:noreply, %{state | retries: 0, connection_pid: pid}}
 
-      {:error, {:already_started, _pid}} ->
-        # Another process already started the connection
-        {:noreply, %{state | retries: 0}}
+      {:error, {:already_started, pid}} ->
+        # This will allow our current node to attempt connections whenever any node's
+        # replication connection dies.
+        Process.link(pid)
```

Comment:
@bmanifold Upon further research, if start_link/3 doesn't succeed, then our pids won't be linked and we should do it explicitly here. That will cause all ReplicationManagers in the cluster to attempt to heal (only one winning) whenever the ReplicationConnection dies.